Lib/test/test_contextlib.py

```diff
@@ -661,6 +661,25 @@ def _exit():
             result.append(2)
         self.assertEqual(result, [1, 2, 3, 4])
 
+    def test_enter_context_errors(self):
+        class LacksEnterAndExit:
+            pass
+        class LacksEnter:
+            def __exit__(self, *exc_info):
+                pass
+        class LacksExit:
+            def __enter__(self):
+                pass
+
+        with self.exit_stack() as stack:
+            with self.assertRaisesRegex(TypeError, 'context manager'):
```

Comment:
If you match "the context manager" then it ensures it's not the async version of the error.

Reply:
Good point!